plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -336,6 +341,20 @@ func (p *pusher) createLogGroupAndStream() error {
 	return err
 }
 
+func (p *pusher) putRetentionPolicy() {
+	if p.Retention > 0 {
+		i := aws.Int64(int64(p.Retention))
+		putRetentionInput := &cloudwatchlogs.PutRetentionPolicyInput{
+			LogGroupName:    &p.Group,
+			RetentionInDays: i,
+		}
+		_, err := p.Service.PutRetentionPolicy(putRetentionInput)
+		if err != nil {
```

Comment:
So we proceed with initializing the agent even if we fail to create the retention policy?

Reply:
right, we don't want to stop the agent from writing logs even though the retention policy setting was not successful